### PR TITLE
Update root readme file

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 1]
```diff
@@ -6,26 +6,38 @@ This library provides access to the Azure Digital Twins service for managing twi
 
 ## Getting started
 
-### Authenticate the Client
+The complete Microsoft Azure SDK can be downloaded from the [Microsoft Azure Downloads][[microsoft_sdk_download]] Page and ships with support for building deployment packages, integrating with tooling, rich command line tooling, and more.
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:9]: Is microsoft_sdk_download supposed to be in double square brackets?

Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:9]: no !

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 2]
```diff
@@ -6,26 +6,38 @@ This library provides access to the Azure Digital Twins service for managing twi
...
 Install the Azure Digital Twins client library for .NET with [NuGet][nuget]:
 
 ```PowerShell
 Install-Package Azure.DigitalTwins.Core
 ```
-### Prerequisites
 
-* An [Azure subscription][azure_sub].
+## Authenticate the Client
+In order to interact with the Azure Digital Twin service, you will need to create an instance of a [ClientSecretCredential][client_secret_credential] and pass it to the constructor of your [DigitalTwinClient][digital_twin_client].
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md]: Not just this type. Any type that derives from TokenCredential. You should include a link to the web docs for the options.

https://docs.microsoft.com/en-us/dotnet/api/azure.core.tokencredential?view=azure-dotnet

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 3]
```diff
@@ -6,26 +6,38 @@ This library provides access to the Azure Digital Twins service for managing twi
...
 
 ## Key concepts
+Azure Digital Twins Preview is an Azure IoT service that creates comprehensive models of the physical environment. It can create spatial intelligence graphs to model the relationships and interactions between people, spaces, and devices.
 
-## Examples
+You can learn more about Azure Digital Twins by visiting [Azure Digital Twins Documentation][digital_twins_documentation]
 
-## Developer guide
+## Examples
+You can familiarize yourself with different APIs using [samples for Digital Twin](./samples/).
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md]: The service name is plural: Digital Twins.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 4]
```diff
@@ -57,8 +69,24 @@ Serialization helpers provided to help serialize/deseralize commonly used types
 Any time the client library code is updated, you will need to run the script [here](./../Export-AdtApis.ps1), which will update the corresponding API surface.
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:69]: Also the Update-Snippets.ps1 script.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md (hunk 3 above)
Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/README.md]: it's already updated by a commit

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 5]
```diff
@@ -57,8 +69,24 @@ Serialization helpers provided to help serialize/deseralize commonly used types
...
 
 ## Troubleshooting
+All Azure Digital Twin service operations will throw a RequestFailedException on failure with helpful ErrorCodes.
+
+For example, if you use the ```GetModelAsync``` operation and the model you are looking for doesn't exist, you can catch that specific [HttpStatusCode][http_status_code] to decide the operation that follows in that case.
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:74]: For inline commands, you just need a single backtick: `GetModelAsync`

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 6]
```diff
@@ -57,8 +69,24 @@ Serialization helpers provided to help serialize/deseralize commonly used types
...
+```csharp
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:75]: Can you tie this to a snippet that gets updated automatically?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 7]
```diff
@@ -6,26 +6,38 @@ This library provides access to the Azure Digital Twins service for managing twi
 
 ## Getting started
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:7]: I recommend getting a markdown linter for VS Code and using it.

One rule us to always leave blank lines around headers, lists, and code blocks.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md (hunk 6 above)
Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:75]: I didn't want to put too much in here, so I just kept it as simple as possible, I think our samples should have enough stuff in them so we don't have to keep updating this, thoughts?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md (hunk 7 above)
Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:7]: I will get it right away and fix the formatting

**Review by drwill-ms — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 8]
```diff
@@ -1,78 +1,134 @@
...
 
 ### /src/Properties
+
 Assembly properties required for running unit tests against signed assemblies in Debug mode.
 
 ### /src/Serialization
-Serialization helpers provided to help serialize/deseralize commonly used types within the digital twins service.
 
-Any time the client library code is updated, you will need to run the script [here](./../Export-AdtApis.ps1), which will update the corresponding API surface.
+Serialization helpers provided to help serialize/deseralize commonly used types within the Digital Twins service.
```
Comment (prmathur-microsoft) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:78]: nit :  deserialize not deseralize

File: sdk/digitaltwins/Azure.DigitalTwins.Core/README.md [hunk 9]
```diff
@@ -1,78 +1,134 @@
 # Azure IoT Digital Twin client library for .NET
```
Comment (prmathur-microsoft) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/README.md:1]: Digital Twins

**Pull request merged**